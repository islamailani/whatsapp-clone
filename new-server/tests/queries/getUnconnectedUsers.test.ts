import { createTestClient } from 'apollo-server-testing'
import { ApolloServer, gql } from 'apollo-server-express'
import schema from '../../schema'
import { users } from '../../db'

describe('Query.unconnectedUsers', () => {
  it('should fetch all users the current user is not chatting with', async () => {
    let currentUser = users[0]

    const server = new ApolloServer({
      schema,
      context: () => ({ currentUser }),
    })

    const { query } = createTestClient(server)

    let res = await query({
      query: gql `
        query GetUnconnectedUsers {
          unconnectedUsers {
            id
            name
            picture
          }
        }
      `,
    })

    expect(res.data).toBeDefined()
    expect(res.errors).toBeUndefined()
    expect(res.data).toMatchSnapshot()

    currentUser = users[1]

    res = await query({
      query: gql `
        query GetUnconnectedUsers {
          unconnectedUsers {
            id
            name
            picture
          }
        }
      `,
    })

    expect(res.data).toBeDefined()
    expect(res.errors).toBeUndefined()
    expect(res.data).toMatchSnapshot()
  })
})
