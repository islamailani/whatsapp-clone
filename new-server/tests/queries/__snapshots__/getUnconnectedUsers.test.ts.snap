// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query.unconnectedUsers should fetch all users the current user is not chatting with 1`] = `
Object {
  "unconnectedUsers": Array [],
}
`;

exports[`Query.unconnectedUsers should fetch all users the current user is not chatting with 2`] = `
Object {
  "unconnectedUsers": Array [
    Object {
      "id": "1",
      "name": "Ray Edwards",
      "picture": "https://randomuser.me/api/portraits/thumb/lego/1.jpg",
    },
    Object {
      "id": "3",
      "name": "Bryan Wallace",
      "picture": "https://randomuser.me/api/portraits/thumb/men/2.jpg",
    },
    Object {
      "id": "4",
      "name": "Avery Stewart",
      "picture": "https://randomuser.me/api/portraits/thumb/women/1.jpg",
    },
    Object {
      "id": "5",
      "name": "Katie Peterson",
      "picture": "https://randomuser.me/api/portraits/thumb/women/2.jpg",
    },
  ],
}
`;
